src/swarm/neo/client/mixins/DeserializeMethod.d

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+
+    Mixin for method to deserialize a received record.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.client.mixins.DeserializeMethod;
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for method to deserialize a received record.
+
+    Params:
+        src = the buffer to be deserialized
+
+*******************************************************************************/
+
+template DeserializeMethod ( alias src )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Deserializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Deserializes `src` into `dst`, using the provided version decorater, if
+        required.
+
+        Params:
+            T = type of struct to deserialize
+            dst = deserialization destination
+
+        Returns:
+            deserialized struct
+
+    ***************************************************************************/
+
+    public Contiguous!(T) deserialize ( T ) ( ref Contiguous!(T) dst )
```

Comment:
I wonder if it would be more convenient for the method to return just `T*` instead, as original ctg instance is anyway reachable at the call site already.